### PR TITLE
fix: [ANDROAPP-3038] Wrong behaviour when selecting a non visible cell in the table

### PR DESCRIPTION
**Pull request by Balcan**
## Description
[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3038)

## Solution description
One a cell is selected, the whole table scrolls to that column position.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [x] Smartphone Emulator
- [ ] Tablet Emulator
- [x] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [x] 7.X
- [x] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: tableview/src/main/java/com/evrencoskun/tableview/handler/SelectionHandler.java [hunk 1]
```diff
@@ -184,10 +184,11 @@ private void selectedCellView() {
 
         // Change background color of the column header which is located on X Position of the cell
         // view.
-        AbstractViewHolder columnHeader = (AbstractViewHolder) mColumnHeaderRecyclerView
+        setHeaderSelection(SelectionState.SHADOWED, shadowColor);
+       /* AbstractViewHolder columnHeader = (AbstractViewHolder) mColumnHeaderRecyclerView
```
Comment (JaimeToca) [inline tableview/src/main/java/com/evrencoskun/tableview/handler/SelectionHandler.java:188]: can you remove the comment?

File: tableview/src/main/java/com/evrencoskun/tableview/handler/SelectionHandler.java [hunk 2]
```diff
@@ -224,10 +250,11 @@ private void unselectedCellView() {
 
         // Change background color of the column header which is located on X Position of the cell
         // view.
-        AbstractViewHolder columnHeader = (AbstractViewHolder) mColumnHeaderRecyclerView
+        setHeaderSelection(SelectionState.UNSELECTED, unSelectedColor);
+       /* AbstractViewHolder columnHeader = (AbstractViewHolder) mColumnHeaderRecyclerView
```
Comment (JaimeToca) [inline tableview/src/main/java/com/evrencoskun/tableview/handler/SelectionHandler.java:254]: comment?

File: app/src/main/java/org/dhis2/data/forms/dataentry/tablefields/edittext/EditTextCellCustomHolder.java [hunk 3]
```diff
@@ -270,7 +274,9 @@ public void dispose() {
     public void selectNext() {
         if (tableView.getColumnHeaderRecyclerView().get(tableView.getColumnHeaderRecyclerView().size() - 1).getAdapter().getItemCount() > tableView.getSelectedColumn() + 1) {
             tableView.setSelectedCell(tableView.getSelectedColumn() + 1, tableView.getSelectedRow());
+//            tableView.scrollToNextField();
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/forms/dataentry/tablefields/edittext/EditTextCellCustomHolder.java:277]: same

File: app/src/main/java/org/dhis2/data/forms/dataentry/tablefields/edittext/EditTextCellCustomHolder.java [hunk 4]
```diff
@@ -71,6 +71,10 @@
                         )
                 );
             }
+
+            if (hasFocus && itemView.getLeft() > 5) {
+                tableView.scrollToColumnPosition(getAdapterPosition(),3);
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/forms/dataentry/tablefields/edittext/EditTextCellCustomHolder.java:76]: what does 3 represent?

File: app/src/main/java/org/dhis2/data/forms/dataentry/tablefields/edittext/EditTextCellCustomHolder.java [hunk 5]
```diff
@@ -71,6 +71,10 @@
                         )
                 );
             }
+
+            if (hasFocus && itemView.getLeft() > 5) {
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/forms/dataentry/tablefields/edittext/EditTextCellCustomHolder.java:75]: can you use constants?

File: tableview/src/main/java/com/evrencoskun/tableview/handler/ScrollHandler.java [hunk 6]
```diff
@@ -93,14 +103,37 @@ private void scrollCellHorizontally(int columnPosition, int offset) {
...
-
         }
     }
 
+    public void scrollToNextField() {
+        final CellRecyclerView child = (CellRecyclerView) mCellLayoutManager.getChildAt(mTableView.getSelectedRow());
+        child.post(new Runnable() {
+            @Override
+            public void run() {
+                child.scrollBy(300, 0);
```
Comment (JaimeToca) [inline tableview/src/main/java/com/evrencoskun/tableview/handler/ScrollHandler.java:114]: can you use smooth scroll??

File: app/src/main/java/org/dhis2/data/forms/dataentry/tablefields/edittext/EditTextCellCustomHolder.java (hunk 4 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/data/forms/dataentry/tablefields/edittext/EditTextCellCustomHolder.java]: It is the default cell cell offset. I'll add a constant.

File: tableview/src/main/java/com/evrencoskun/tableview/handler/ScrollHandler.java (hunk 6 above)
Comment (Balcan) [reply tableview/src/main/java/com/evrencoskun/tableview/handler/ScrollHandler.java:114]: Nope. Smooth scroll is messing with all the scroll listeners

File: app/src/main/java/org/dhis2/data/forms/dataentry/tablefields/edittext/EditTextCellCustomHolder.java (hunk 5 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/data/forms/dataentry/tablefields/edittext/EditTextCellCustomHolder.java]: removed in fact :)

**Pull request merged**